docs/kbs_attestation_protocol.md

```diff
@@ -108,113 +146,303 @@ attestation challenge to KBC. The payload format is as follows:
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ┃                ┌───────────────────┐              ┃
+     ┃────────────────┼POST /kbs/v0/attest┼─────────────▶┃
+     ┃        ┌───────┴───────────────────┴───────┐      ┃
+     ┃        │Header: "Cookie kbs-session-id=123"│      ┃
+     ┃        │Body: Attestation                  │      ┃
+     ┃        └───────────────────────────────────┘      ┃
+     ┃                     ┌────────┐                    ┃
+     ┃◀────────────────────┼Response┼────────────────────┃
+     ┃      ┌──────────────┴────────┴───────────────┐    ┃
+     ┃      │Header: "Set-Cookie kbs-session-id=123"│    ┃
+     ┃      │Body: N/A                              │    ┃
+     ┃      └───────────────────────────────────────┘    ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ▽                                                   ▽
+               Figure 1: KBS Authentication Phase
+
+```
+
+## Resource Requests
+
+The [authentication](#authentication) phase is the first step of the overall KBS
+protocol, where attesters authenticate themselves against a KBS implementation.
+This leads to the second phase, during which authenticated attesters can request
+protected resources from the KBS.
+
+The KBS implementation keeps track of attestation results and binds them to a
+cookie identifier. During the second phase, the KBS can then decide if a
+specific resource could be released to a given attester, by mapping the cookie
+identifier the attester includes in its resource request message to its
+attestation results.
+
+To request a protected resource from the KBS, the attester sends a `GET` request
+to a resource specific endpoint. If the attester is allowed to access the
+resource, the KBS will respond to the `GET` request with an HTTP response which
+content is set to a [KBS Response](#response) JSON payload:
+
+```
+┌──────────┐                                         ┌─────┐
+│ Attester │                                         │ KBS │
+└───┳──────┘                                         └──┳──┘
+    ┃            ┌─────────────────────────┐            ┃
+    ┃────────────┼GET /kbs/v0/<resource_id>┼───────────▶┃
+    ┃      ┌─────┴─────────────────────────┴────┐       ┃
+    ┃      │Header: "Cookie kbs-session-id=123" │       ┃
+    ┃      │Body: N/A                           │       ┃
+    ┃      └────────────────────────────────────┘       ┃
+    ┃                    ┌────────┐                     ┃
+    ┃◀───────────────────┼Response┼─────────────────────┃
+    ┃     ┌──────────────┴────────┴───────────────┐     ┃
+    ┃     │Header: "Set-Cookie kbs-session-id=123"│     ┃
```

Comment:
How does the KBS expire these session IDs to avoid having to cache them forever (aka a potential slow memory leak)? There doesn't appear to be a final call that the Attester makes. But even if there was that wouldn't be sufficient to deal with a buggy / nefarious attester. I'm assuming some sort of timeout, but where would that need to be specified / documented?

Reply:
I think it's reasonable leaving that up to the implementation and out of the scope of the protocol. Managing the lifetime of the sessions is a extremely common thing in web development and there is plenty of literature about best practices.

Reply:
Maybe we could note explicitly that this is out of scope to avoid any confusion then? I would imagine that a very long expiration time could be problematic from a memory and security standpoint.